From 6dfa733cabd717b6f17c9d1d4985c6a46f291db4 Mon Sep 17 00:00:00 2001
From: Scott James Remnant <scott@netsplit.com>
Date: Thu, 3 Apr 2014 15:22:37 -0700
Subject: [PATCH 2/5] core: don't try BR/EDR for LE-only devices

Calling Connect() after Pair() should not fail for LE-only devices.
---
 src/device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/device.c b/src/device.c
index 56b1780..289d522 100644
--- a/src/device.c
+++ b/src/device.c
@@ -1425,7 +1425,7 @@ static DBusMessage *dev_connect(DBusConnection *conn, DBusMessage *msg,
 
 	if (dev->bredr_state.connected)
 		bdaddr_type = dev->bdaddr_type;
-	else if (dev->le_state.connected)
+	else if (dev->le_state.connected && dev->bredr)
 		bdaddr_type = BDADDR_BREDR;
 	else
 		bdaddr_type = select_conn_bearer(dev);
-- 
1.9.1.423.g4596e3a

